### PR TITLE
`azurerm_key_vault` - use `virtual_network_rules` as block and add support for `ignore_missing_vnet_service_endpoint` property

### PR DESCRIPTION
**Pull request by wuxu92**
**Looks like there is a problem in updating `ignore_missing_vnet_service_endpoint` property. it takes no effect on the exists keyvault resource**, so convert this PR to draft status. created an issue to track this: https://github.com/Azure/azure-rest-api-specs/issues/22128

`ignore_missing_vnet_service_endpoint` need support from version `2021-10-01`, and mark `virtual_network_subnet_ids` as deprecated.

https://github.com/Azure/azure-rest-api-specs/blob/63a4b37b66740087b113e00c1b32b2b9c98fd9db/specification/keyvault/resource-manager/Microsoft.KeyVault/stable/2021-10-01/keyvault.json#L1798-L1800

```
--- PASS: TestAccDataSourceKeyVault_networkAcls (543.73s)
```

### CODE REVIEW

File: website/docs/r/key_vault.html.markdown [hunk 1]
```diff
@@ -141,7 +141,17 @@ A `network_acls` block supports the following:
...
+
+* `virtual_network_rules` - (Optional) One or more `virtual_network_rules` blocks as defined below.
+
+---
+
+A `virtual_network_rules` block supports the following:
+
+* `id` - (Required) Specifies the subnet ID for this rule.
+
+* `ignore_missing_vnet_service_endpoint_enabled` - (Optional) Whether NRP will ignore the check if parent subnet has serviceEndpoints configured.
```
Comment (jackofallops) [inline website/docs/r/key_vault.html.markdown:154]: Can we make this clearer for users, the description straight from the Swagger isn't very helpful for new users?

**Review by jackofallops — CHANGES_REQUESTED** [review]
Hi @wuxu92 - Thanks for this PR.
I've left some comments below, if you can take a look?

Thanks

File: internal/services/keyvault/key_vault_resource.go [hunk 2]
```diff
@@ -178,11 +182,36 @@ func resourceKeyVault() *pluginsdk.Resource {
...
+							Type:          pluginsdk.TypeSet,
+							Optional:      true,
+							Computed:      true,
+							Deprecated:    "virtual_network_rules is preferred",
+							ConflictsWith: []string{"network_acls.virtual_network_rules"},
+							Elem:          &pluginsdk.Schema{Type: pluginsdk.TypeString},
+							Set:           set.HashStringIgnoreCase,
+						},
+
+						"virtual_network_rules": {
```
[the remark this answers is not in the document]
Comment (wuxu92) [reply internal/services/keyvault/key_vault_resource.go:196]: Should it rename `virtual_network_rules` to `virtual_network_subnet_id`?

I have tested several times locally and the order is stable. it doesn't need a `TypeSet` here.

**Comment by katbyte** [comment]
@wuxu92 - is this ready for a review or still a draft?

**Comment by wuxu92** [comment]
close this for swagger issue.

**Pull request closed without merging**